32/33/34/36 Updates Component for compatibility in existing projects.

samrocksc opened the pull request:
#32 #33 #34 #36

samrocksc committed:
Update README with new changes.

samrocksc committed:
Squashed Commits

chrisshiplet commented on src/DraftJSEditor.jsx:
```diff
@@ -13,6 +13,8 @@ import {
 
 import linkDecorator from './decorators/link';
 
+import '../lib/draftjs-editor.min.css';
```
IIRC draftjs-editor assumes the app which depends on it will include the CSS itself if desired. This way we give end users the choice between importing/tweaking the Stylus styles or using the minified CSS directly.

chrisshiplet commented on src/DraftJSEditor.jsx:
```diff
@@ -349,13 +351,33 @@ class DraftJSEditor extends Component {
 
+    const passedProps = this.props;
+
+    const cleanObject = (object, keyArray) => {
```
This feels like something that would be a good idea to break out into its own file, something like `src/utils/cleanObject.js`

chrisshiplet commented on src/DraftJSEditor.jsx:
```diff
@@ -349,13 +351,33 @@ class DraftJSEditor extends Component {
+      return newObj;
+    }
+
+    const passableProps = cleanObject(passedProps, [
```
If I'm understanding what's going on here at a glance, I think something along these lines would accomplish the same?

`const { content, customBlockControls, customBlocks, inlineControls, controlDisplay, containerStyle, ...passableProps } = this.props;`

With the destructuring syntax you're assigning those first 6 things as `const`s and then the rest of the object will get dumped in `passableProps`. Then you might not need that utility function above?

chrisshiplet commented on storybook/stories/SetStateExample.js:
```diff
@@ -38,6 +43,7 @@ class SetStateExample extends Component {
           <a href="https://github.com/synapsestudios/draftjs-editor" target="_blank">Github</a>
         </p>
         <DraftJSEditor
+          containerStyle={saneStyling}
```
I think what #34 was getting at was that projects shouldn't need to provide these themselves, but rather this library would have sane defaults. In this case we're still relying on the project to pass `saneStyling`, so we should probably add these styles to the CSS for `.DraftJSEditor-root` directly.

samrocksc committed:
Updates with changes made.
1. Creates defaults for height/width of component in CSS.
2. Uses destructuring to create the `passableProps`.
3. Updates the example demo with a more tolerable name for container
style.
4. Updates component to force mandatory inclusion of styles.
